iam-persistence/src/main/java/it/infn/mw/iam/persistence/model/IamX509Certificate.java

```diff
@@ -102,12 +102,7 @@ public boolean equals(Object obj) {
     if (getClass() != obj.getClass())
       return false;
     IamX509Certificate other = (IamX509Certificate) obj;
-    if (subjectDn == null) {
-      if (other.subjectDn != null)
-        return false;
-    } else if (!subjectDn.equals(other.subjectDn))
-      return false;
-    return true;
+    return issuerDn.equals(other.issuerDn);
```

Comment:
Equality should consider both issuer and subject. Apart from the null checks, this should be something like `return subjectDn.equals(other.subjectDn) && issuerDn.equals(other.issuerDn);`.

Reply:
We need to change also `hashCode()`, which has to consider both issuer and subject. The implementation can rely on `Objects.hashCode()` or, even better, `Objects.hash()`, i.e. `return Objects.hash(subjectDn, issuerDn);`. See for example https://www.baeldung.com/java-objects-hash-vs-objects-hashcode.